### PR TITLE
chore: update baseline to OTEL 1.15.0 and remove unused dependencies

### PR DESCRIPTION
**Pull request by mmanciop**

### CODE REVIEW

**Review by GuyMoses — APPROVED** [review]

File: .github/workflows/push-actions.yml [hunk 1]
```diff
@@ -1,12 +1,38 @@
...
+      - name: Install jq
+        run:
+          sudo apt-get update && sudo apt-get install -y jq
+      - name: Install Python dependencies
+        run: |
+          pip install -r requirements.txt
+      - name: Generate list of integration tests
+        id: discover_itests
+        run: |
+          echo "integration_tests=$(nox --list | grep 'integration_tests_' | awk '{ print $2 }' | sed -r 's|^integration_tests_([^-]*)(-[[:digit:]].[[:digit:]]+)?\(.*\)$|\1|' | uniq | sort | jq -Rnc '[inputs]')" | tee /dev/stderr >> $GITHUB_OUTPUT
```
Comment (GuyMoses) [inline .github/workflows/push-actions.yml:25]: can you re-write these lines in a more readable way?

File: src/test/integration/boto3/tests/test_boto3.py [hunk 2]
```diff
@@ -15,17 +15,17 @@ def assert_is_version(self, version: str):
         self.assertTrue(patch.isdigit())
 
     def assert_have_env_vars(self, root: dict):
-        collected_envs: str = root["resource"]["process.environ"]
+        collected_envs: str = root["resource"]["attributes"]["process.environ"]
```
Comment (GuyMoses) [inline src/test/integration/boto3/tests/test_boto3.py:18]: do we need to change anything in the BE?

Comment (mmanciop) [reply src/test/integration/boto3/tests/test_boto3.py:18]: No need of BE changes, this is not a change in the OTLP we export to SaaS, it is a change in the format of the `ConsoleExporter` that makes it look more like actual OTLP.

File: .github/workflows/push-actions.yml (hunk 1 above)
Comment (mmanciop) [reply .github/workflows/push-actions.yml]: a61902668722bca37da604878c3c497c8eae5b53 is a masterpiece, if I say so myself :D

**Review by GuyMoses — APPROVED** [review]

**Pull request merged**